ow_power_system/include/PowerSystemPack.h

```diff
@@ -0,0 +1,134 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef __POWER_SYSTEM_PACK_H__
+#define __POWER_SYSTEM_PACK_H__
+
+#include <vector>
+#include <map>
+#include <chrono>
+#include <random>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include <PrognoserFactory.h>
+#include "PowerSystemNode.h"
+
+#include "Messages/MessageBus.h"
+#include "Messages/ProgEventMessage.h"
+#include "Messages/ScalarMessage.h"
+#include "ModelBasedAsyncPrognoserBuilder.h"
+
+using PrognoserMap = std::map<PCOE::MessageId, PCOE::Datum<double>>;
+using PrognoserVector = std::vector<PrognoserMap>;
+
+// Change this value to modify the number of parallel battery nodes in the pack.
+// NOTE: The expected number of nodes (24) appears to take much longer than 2
+//       seconds to process. Currently unknown what the cause of this is, though
+//       it is likely related to GSAP's time required to process predictions and
+//       so it may not be addressable here.
+const int NUM_NODES = 8;
+
+// Struct that contains the information used for EoD predictions.
+struct ModelInfo {
+  double timestamp;
+  double wattage;
+  double voltage;
+  double temperature;
+};
+
+// Struct that groups the variables/classes used to handle PowerSystemNodes.
+struct PowerNode {
+  std::string name;
+  PowerSystemNode node;
+  MessageBus bus;
+  ModelInfo model;
+  double previous_time;
+};
+
+// Struct that groups the end-of-discharge (EoD) prediction values.
+struct EoDValues {
+  double remaining_useful_life;
+  double state_of_charge;
+  double battery_temperature;
+};
+
+class PowerSystemPack
+{
+public:
```

Comment:
Good to be explicit about the copy constructor and assignment operator, .e.g. define them or use `= delete` or `= default`.  Same for the destructor, though either define or default it, since it's required.